### ISSUE TITLE
在使用model中 forward函数self.fc1(x)报错（obs为离散型）

### ISSUE THREAD

**Issue by Saber-xxf** [PaddlePaddle/PARL#763]
![image](https://user-images.githubusercontent.com/71160585/144822488-d4b979af-082e-4db2-9d1b-efd6cce2352d.png)
代码如下：其中obs是离散的 obs = paddle.to_tensor(obs, dtype='int32')
![image](https://user-images.githubusercontent.com/71160585/144822613-eb3461d7-81cb-4564-8b97-dd220c220552.png)
而当我把 obs = paddle.to_tensor(obs, dtype='float32')时，报错
![image](https://user-images.githubusercontent.com/71160585/144831885-d22bc779-f094-4536-b8d6-1b939a1d1786.png)
使用的 parl,和paddle都是最新版，env是自己写的 self.observation_space = spaces.Discrete(15)

**Comment by TomorrowIsAnOtherDay** [comment]
请问你跑的代码是仓库中的哪份代码呢？是否有改动过呢？

**Comment by Saber-xxf** [comment]
跑的DQN,代码 只是换了自己的环境env，自己的环境用q-learning表格，这次用了神经网络唯一的区别就是自己的环境值obs为离散型的range(1,17)

**Comment by TomorrowIsAnOtherDay** [comment]
了解了，看起来像是你的神经网络参数尺寸设置不对，你的obs输入维度是17，那你的fc1的维度是多少呢？

**Comment by Saber-xxf** [comment]
self.states = range(1,17)这是obs ，fc1不是直接调用的吗 
`
hid1_size = act_dim * 10
self.fc1 = nn.Linear(obs_dim, hid1_size)
self.fc2 = nn.Linear(hid1_size, act_dim)
self.tanh = nn.Tanh()
self.softmax = nn.Softmax()

 def forward(self, x):
        out = self.tanh(self.fc1(x))
        prob = self.softmax(self.fc2(out))

        return prob

**Comment by Saber-xxf** [comment]
> 了解了，看起来像是你的神经网络参数尺寸设置不对，你的obs输入维度是17，那你的fc1的维度是多少呢？

hid1_size = act_dim * 10
self.fc1 = nn.Linear(obs_dim, hid1_size)
self.fc2 = nn.Linear(hid1_size, act_dim)
self.tanh = nn.Tanh()
self.softmax = nn.Softmax()

**Comment by TomorrowIsAnOtherDay** [comment]
不是直接调用一个接口这么简单的。你定义好state之后，调用fc实际上是和fc内部的矩阵做乘法，你的fc 参数尺寸设置的都不对，这个矩阵没法和你的state相乘。

**Comment by xxf-liub** [comment]
> 

这个应该在哪里修改，没看到相关的文档

**Comment by Saber-xxf** [comment]
> 这个应该在哪里修改，没看到相关的文档



> 不是直接调用一个接口这么简单的。你定义好state之后，调用fc实际上是和fc内部的矩阵做乘法，你的fc 参数尺寸设置的都不对，这个矩阵没法和你的state相乘。

这个应该在哪里修改，没看到相关的文档

**Comment by TomorrowIsAnOtherDay** [comment]
这个是基础的神经网络概念，并非强化学习的基础知识。
建议先去熟悉下神经网络，做一下简单的回归拟合，再回来解决这个问题。

**Comment by Saber-xxf** [comment]
> 


> 这个是基础的神经网络概念，并非强化学习的基础知识。 建议先去熟悉下神经网络，做一下简单的回归拟合，再回来解决这个问题。

ok谢谢了
